### PR TITLE
Add SampleSet.to_pandas_dataframe method

### LINKED ISSUE #332: Export SampleSet object to Pandas df
**Application**
It would be useful for Pandas users to be able to export the `SampleSet`/`Response` objects into a pandas dataframe.

**Proposed Solution**
```
samples = dimod.ExactSolver().sample_ising(h, J)
df = samples.to_pandas_dataframe()
```

### PR DESCRIPTION
**Pull request by arcondello**
Closes #332

### CODE REVIEW

**Comment by arcondello** [comment]
Closes #334

File: tests/test_sampleset.py [hunk 1]
```diff
@@ -158,3 +165,16 @@ def test_functional_str(self):
...
+class TestSampleSet_to_pandas_dataframe(unittest.TestCase):
+    def test_simple(self):
+        samples = dimod.SampleSet.from_samples(([[-1, 1, -1], [-1, -1, 1]], 'abc'),
+                                               dimod.SPIN, energy=[-.5, .5])
+        df = samples.to_pandas_dataframe()
+
+        other = pd.DataFrame([[-1, 1, -1, -.5, 1], [-1, -1, 1, .5, 1]],
+                             columns=['a', 'b', 'c', 'energy', 'num_occurrences'])
+
+        pd.testing.assert_frame_equal(df, other, check_dtype=False)
```
Comment (jberwald) [inline tests/test_sampleset.py]: Please consider adding a test for duplicate samples. When these occur they will produce duplicate rows in the DF. For example if `sample1 = {'a':1, 'b':0}` is sampled  twice by QC it should show up in the sample set with  `num_occurrences =2`. I find that this is not always true (no clue why), and that I sometimes get two separate instances of `sample1` in a returned Response object. I have resorted to a groupby in pandas to account for this and get the correct `num_occurrences`.

Comment (arcondello) [reply tests/test_sampleset.py]: I think the best approach would be to add an aggregator to the `SampleSet` object that can be invoked before `to_pandas_dataframe`. See #268
